test/governance/ATMTokenTest.js

```diff
@@ -0,0 +1,190 @@
+    },function(
+        newCap,
+        sender,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or should not be able to set cap correctly', mustFail), async function() {
+
+            try {
+                // Invocation
+                await instance.setCap(newCap, { from: sender });
+                const result = await instance.cap();
+                // Assertions
+                assert(!mustFail, 'It should have failed because the sender is invalid');
+                assert(result);
+            } catch (error) {
+                // Assertions
+                assert(mustFail);
+                assert(error);
+                assert.equal(
+                    error.reason,
+                    expectedErrorMessage
+                    );
+            }
+
+        });
+    });
+
+    withData({
+        _10_claim_vested_basic: [daoMember2, 1000, 7000, 7001, undefined, false],
+        _11_claim_vested_before_deadline: [daoMember2, 1000, 7000, 5000, 'Vesting deadline has not passed', true],
+        _12_claim_vested_no_amount: [daoMember1, 1000, 7000, 8000, 'Account does not have a vesting balance!', true]
+    },function(
+        receipent,
+        amount,
+        vestingPeriod,
+        claimTime,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or should not be able to claim correctly', mustFail), async function() {
+
+            try {
+                // Invocation
+                await instance.mintVesting(daoMember2, amount, vestingPeriod, { from: daoAgent });
+                const currentTime = await timer.getCurrentTimestampInSeconds();
+                await timer.advanceBlockAtTime(currentTime + claimTime);
+                const result = await instance.withdrawVested({ from: receipent });
+                // Assertions
+                assert(!mustFail, 'It should have failed because the account is not vested');
+                assert(result);
+            } catch (error) {
+                // Assertions
+                assert(mustFail);
+                assert(error);
+                assert.equal(
+                    error.reason,
+                    expectedErrorMessage
+                    );
+            }
+
+        });
+    });
+
+    withData({
+        _13_revoke_vested_basic: [daoMember1, 1000, 7000, undefined, false],
+        _14_revoke_vested_no_amount: [daoMember2, 1000, 7000, "Account does not have a vesting balance!", true]
+        
+    },function(
+        receipent,
+        amount,
+        vestingPeriod,
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('agent', 'mint', 'Should or be able to revoke correctly', mustFail), async function() {
+
+            try {
+                // Invocation
+                await instance.mintVesting(daoMember1, amount, vestingPeriod, { from: daoAgent });
```

Comment:
I think if you are testing the revokeVesting function, the mintVesting should be before the execution comment:

```
// Setup
await instance.mintVesting(daoMember1, amount, vestingPeriod, { from: daoAgent });

try {
                // Invocation
                const result = await instance.revokeVesting(receipent, { from: daoAgent });

                // Assertions
                ...

```

Similar in other unit tests.

Reply:
Changed